pkg/git/libgit2/checkout.go

```diff
@@ -170,70 +225,106 @@ type CheckoutTag struct {
 func (c *CheckoutTag) Checkout(ctx context.Context, path, url string, opts *git.AuthOptions) (_ *git.Commit, err error) {
 	defer recoverPanic(&err)
 
-	remoteCallBacks := RemoteCallbacks(ctx, opts)
-	proxyOpts := &git2go.ProxyOptions{Type: git2go.ProxyTypeAuto}
-
-	repo, remote, err := initializeRepoWithRemote(ctx, path, url, opts)
-	if err != nil {
-		return nil, err
-	}
-	// Open remote connection.
-	err = remote.ConnectFetch(&remoteCallBacks, proxyOpts, nil)
-	if err != nil {
-		remote.Free()
-		repo.Free()
-		return nil, fmt.Errorf("unable to fetch-connect to remote '%s': %w", managed.EffectiveURL(url), gitutil.LibGit2Error(err))
-	}
-	defer func() {
-		remote.Disconnect()
-		remote.Free()
-		repo.Free()
-	}()
-
-	// When the last observed revision is set, check whether it is still the
-	// same at the remote branch. If so, short-circuit the clone operation here.
-	if c.LastRevision != "" {
-		heads, err := remote.Ls(c.Tag)
+	// This branching is temporary, to address the transient panics observed when using unmanaged transport.
+	// The panics probably happen because we perform multiple fetch ops (introduced as a part of optimizing git clones).
+	// The branching lets us establish a clear code path to help us be certain of the expected behaviour.
+	// When we get rid of unmanaged transports, we can get rid of this branching as well.
+	if managed.Enabled() {
+		if opts.TransportOptionsURL == "" {
```

Comment:
In `CheckoutBranch`, a nil check is performed for this option. That's missing here.